### PR TITLE
add use_shared_memory option

### PR DESCRIPTION
**Pull request by shu65**
This PR adds `use_shared_memory` to share data among agents.
When I try to another method to share data, I got errors.

To solve it,  I would like to add the option in `train_agent_async()`

### CODE REVIEW

File: pfrl/experiments/train_agent_async.py [hunk 1]
```diff
@@ -207,11 +208,11 @@ def train_agent_async(
             other thread raised an excpetion. The train will be terminated and
             the current agent will be saved.
             If set to None, a new Event object is created and used internally.
+        use_shared_memory (bool): Prevent using shared memory.
```
Comment (prabhatnagarajan) [inline pfrl/experiments/train_agent_async.py:211]:
```suggestion
        use_shared_memory (bool): Share memory amongst asynchronous agents.
```
If true, then share memory, right? So we shouldn't say "Prevent"?

**Review by prabhatnagarajan — CHANGES_REQUESTED** [review]

**Comment by shu65** [comment]
Thank you for the comments. I fixed the docstring.

**Review by muupan — APPROVED** [review]

**Pull request merged**